mlflow/utils/file_utils.py

```diff
@@ -186,6 +186,17 @@ def read_yaml(root, file_name):
         raise e
 
 
+def dump_yaml(data):
+    """
+    Dump data from a dictionary into a formatted yaml
+
+    :param contents: Dictionary containing yaml contents
+
+    :return: Formatted yaml containing data
+    """
+    return yaml.dump(data)
```

Comment:
We already have a `write_yaml` util in this file. Can we use that instead (after augmenting for append capabilities as suggested in https://github.com/mlflow/mlflow/pull/6752/files#r968847408)?

Reply:
here i am using `dump_yaml` for outputting data as yaml (rather than as a python dict) for the step card. i can't use `write_yaml` because that writes to a file instead.